pkg/pgclient/client.go

```diff
@@ -297,3 +309,43 @@ func observeStatementCacheState(conn *pgx.Conn) bool {
 	statementCacheLen.Observe(float64(statementCacheSize))
 	return true
 }
+
+func healthCheckRoutine(ctx context.Context, conn pgxconn.PgxConn) {
+	r := prometheus.DefaultRegisterer
+	if env := os.Getenv("IS_TEST"); env == "true" {
+		r = prometheus.NewRegistry()
+	}
+	dbHealthChecks := prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "database",
+			Name:      "health_checks_total",
+			Help:      "Total number of database health checks performed.",
+		},
+	)
+	dbHealthErrors := prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "database",
+			Name:      "health_check_errors_total",
+			Help:      "Total number of database health check errors.",
+		},
+	)
+	r.MustRegister(dbHealthChecks, dbHealthErrors)
+	go func() {
+		check := time.NewTicker(time.Minute)
+		defer check.Stop()
+		connection := health.NewHealthChecker(conn)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-check.C:
+			}
+			dbHealthChecks.Inc()
+			if err := connection(); err != nil {
```

Comment:
Since not being able to communicate to DB is a major failure I believe it's good to log error as well.

Reply:
This is now removed to #1193 as this metric suits being part of that PR.